### PR TITLE
fix: Handle templates with invalid strftime elements

### LINKED ISSUE #2736: Template from invalid strftime
Currently if string for `Template` have [Item::Error](https://docs.rs/chrono/0.4.6/chrono/format/enum.Item.html#variant.Error) (added in https://github.com/timberio/vector/pull/1003) nothing will happened. As result issues like https://github.com/timberio/vector/issues/2729 arise.
Proper way will be return `crate::Result<Template>` from `Template::from`, but for start maybe add `panic!` at least?

### PR DESCRIPTION
**Pull request by bruceg**
Please make sure I did all the error handling correctly. There were a couple of cases where I guessed at the right error handling result.

Closes #2736 

Signed-off-by: Bruce Guenter <bruce@timber.io>

### CODE REVIEW

File: src/template.rs [hunk 1]
```diff
@@ -29,20 +31,53 @@ pub struct Template {
...
+            .filter(|item| matches!(item, Item::Error))
+            .count()
+        {
+            0 => Ok(Template {
+                src: src.into(),
+                src_bytes: src.into(),
+                has_ts: StrftimeItems::new(src).filter(is_dynamic).count() > 0,
+                has_fields: RE.is_match(src),
+            }),
+            _ => Err(TemplateError::StrftimeError),
```
Comment (fanatid) [inline src/template.rs:63]: `StrftimeItems::new(src)` called twice, i.e. `src` also will be parsed twice.
Instead `.filter(|item| matches!(item, Item::Error)).count()` [.any](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.any) can be used.

**Comment by fanatid** [comment]
Thank you Bruce, rest things looks good!

File: src/template.rs (hunk 1 above)
Comment (bruceg) [reply src/template.rs:63]: Yeah, that was not my finest piece of code, just wanted to get it written. I'll see how I can rework it.

File: src/sinks/gcp/cloud_storage.rs [hunk 2]
```diff
@@ -197,11 +200,12 @@ impl GcsSink {
 
         let batch = config.batch.unwrap_or(bytesize::mib(10u64), 300);
 
-        let key_prefix = if let Some(kp) = &config.key_prefix {
-            Template::from(kp.as_str())
-        } else {
-            Template::from("date=%F/")
-        };
+        let key_prefix = config
```
Comment (Hoverbear) [inline src/sinks/gcp/cloud_storage.rs:203]: It seems like lines 200-204 already do similar? Is one redundant?

**Review by fanatid — APPROVED** [review]

**Review by Hoverbear — APPROVED** [review]

**Review by lukesteensen — APPROVED** [review]

**Pull request merged**